// @jjuidev/node-devtools/dist/cli/cli.cjs
#!/usr/bin/env node
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));

// src/scripts/cli.ts
var import_commander = require("commander");

// src/scripts/interactive-setup.ts
var import_consola2 = __toESM(require("consola"), 1);
var import_picocolors2 = __toESM(require("picocolors"), 1);
var import_prompts = __toESM(require("prompts"), 1);

// src/utils/package-lists.ts
var BASE_PACKAGES = [
  // pkgs common to husky and lint-staged
  {
    name: "husky",
    dev: true
  },
  {
    name: "lint-staged",
    dev: true
  },
  // pkgs common to commitlint
  {
    name: "@commitlint/cli",
    dev: true
  },
  {
    name: "@commitlint/config-conventional",
    dev: true
  },
  // pkgs common to eslint and prettier
  {
    name: "prettier",
    dev: true
  },
  {
    name: "eslint",
    dev: true
  },
  {
    name: "@eslint/js",
    dev: true
  },
  {
    name: "globals",
    dev: true
  },
  {
    name: "typescript-eslint",
    dev: true
  }
];
var ESLINT_BASE_PACKAGES = [
  {
    name: "eslint-config-prettier",
    dev: true
  },
  {
    name: "eslint-plugin-autofix",
    dev: true
  },
  {
    name: "eslint-plugin-import",
    dev: true
  },
  {
    name: "eslint-plugin-prefer-arrow-functions",
    dev: true
  }
];
var PRETTIER_BASE_PACKAGES = [];
var CONDITIONAL_PACKAGES = [
  {
    name: ["eslint-plugin-react", "eslint-plugin-react-hooks", "eslint-plugin-react-native"],
    dev: true,
    condition: (answers) => answers.framework === "react" || answers.framework === "next" || answers.framework === "react-native"
  },
  {
    name: "@next/eslint-plugin-next",
    dev: true,
    condition: (answers) => answers.framework === "next"
  },
  {
    name: ["eslint-plugin-tailwindcss", "prettier-plugin-tailwindcss"],
    dev: true,
    condition: (answers) => answers.useTailwind
  },
  {
    name: "eslint-plugin-storybook",
    dev: true,
    condition: (answers) => answers.useStorybook
  }
];
var getPackagesToInstall = (answers) => {
  const packages = [];
  BASE_PACKAGES.forEach((pkg) => {
    const names = Array.isArray(pkg.name) ? pkg.name : [pkg.name];
    packages.push(...names);
  });
  ESLINT_BASE_PACKAGES.forEach((pkg) => {
    const names = Array.isArray(pkg.name) ? pkg.name : [pkg.name];
    packages.push(...names);
  });
  PRETTIER_BASE_PACKAGES.forEach((pkg) => {
    const names = Array.isArray(pkg.name) ? pkg.name : [pkg.name];
    packages.push(...names);
  });
  CONDITIONAL_PACKAGES.forEach((pkg) => {
    if (pkg.condition && pkg.condition(answers)) {
      const names = Array.isArray(pkg.name) ? pkg.name : [pkg.name];
      packages.push(...names);
    }
  });
  return packages;
};

// src/utils/package-manager.ts
var import_child_process = require("child_process");
var import_fs = require("fs");
var import_path = require("path");
var detectPackageManager = () => {
  const cwd = process.cwd();
  if ((0, import_fs.existsSync)((0, import_path.join)(cwd, "bun.lockb"))) {
    return "bun";
  }
  if ((0, import_fs.existsSync)((0, import_path.join)(cwd, "pnpm-lock.yaml"))) {
    return "pnpm";
  }
  if ((0, import_fs.existsSync)((0, import_path.join)(cwd, "yarn.lock"))) {
    return "yarn";
  }
  return "npm";
};

// src/scripts/setup-typescript-alias.ts
var import_fs2 = require("fs");
var import_path2 = require("path");
var import_consola = __toESM(require("consola"), 1);
var import_picocolors = __toESM(require("picocolors"), 1);
var setupTypescriptAlias = () => {
  import_consola.default.start("Setting up TypeScript alias imports...");
  const tsconfigPath = (0, import_path2.join)(process.cwd(), "tsconfig.json");
  if (!(0, import_fs2.existsSync)(tsconfigPath)) {
    import_consola.default.warn("tsconfig.json not found. Skipping TypeScript alias setup.");
    return;
  }
  try {
    const tsconfigContent = (0, import_fs2.readFileSync)(tsconfigPath, "utf8");
    const tsconfig = JSON.parse(tsconfigContent);
    if (!tsconfig.compilerOptions) {
      tsconfig.compilerOptions = {};
    }
    tsconfig.compilerOptions.baseUrl = "src";
    tsconfig.compilerOptions.paths = {
      "@/*": ["./*"]
    };
    (0, import_fs2.writeFileSync)(tsconfigPath, JSON.stringify(tsconfig, null, "	"), "utf8");
    import_consola.default.success(`Updated ${import_picocolors.default.cyan("tsconfig.json")} with alias imports:
`);
    import_consola.default.info(`  ${import_picocolors.default.gray("\u2022")} baseUrl: ${import_picocolors.default.green('"src"')}`);
    import_consola.default.info(`  ${import_picocolors.default.gray("\u2022")} paths: ${import_picocolors.default.green('{ "@/*": ["./*"] }')}
`);
  } catch (error) {
    import_consola.default.error("Failed to update tsconfig.json");
    if (error instanceof Error) {
      import_consola.default.error(error.message);
    }
  }
};

// src/scripts/interactive-setup.ts
var runInteractiveSetup = async () => {
  import_consola2.default.box({
    title: "\u{1F6E0}\uFE0F  Node DevTools Setup",
    message: "Interactive configuration for your development environment",
    style: {
      borderColor: "cyan",
      borderStyle: "rounded"
    }
  });
  import_consola2.default.info("Let's configure your development tools!\n");
  const answers = await (0, import_prompts.default)(
    [
      {
        type: "select",
        name: "framework",
        message: "\u{1F680} Which framework do you use?",
        choices: [
          { title: "\u{1F4E6} Node.js", value: "node" },
          { title: "\u269B\uFE0F  React", value: "react" },
          { title: "\u{1F4F1} React Native", value: "react-native" },
          { title: "\u25B2  Next.js", value: "next" }
        ],
        initial: 0
      },
      {
        type: (prev, values) => values.framework === "react" || values.framework === "next" ? "confirm" : null,
        name: "useTailwind",
        message: "\u{1F3A8} Are you using Tailwind CSS?",
        initial: false
      },
      {
        type: (prev, values) => values.framework === "react" || values.framework === "next" ? "confirm" : null,
        name: "useStorybook",
        message: "\u{1F4DA} Are you using Storybook?",
        initial: false
      },
      {
        type: "confirm",
        name: "useTypescriptAlias",
        message: "\u{1F4C2} Do you want to use TypeScript alias imports?",
        initial: false
      }
    ],
    {
      onCancel: () => {
        import_consola2.default.warn("\nSetup cancelled by user");
        process.exit(0);
      }
    }
  );
  const setupAnswers = {
    framework: answers.framework || "node",
    useTailwind: answers.useTailwind || false,
    useStorybook: answers.useStorybook || false,
    useTypescriptAlias: answers.useTypescriptAlias || false
  };
  const frameworkLabels = {
    node: "\u{1F4E6} Node.js",
    react: "\u269B\uFE0F  React",
    "react-native": "\u{1F4F1} React Native",
    next: "\u25B2  Next.js"
  };
  const summaryLines = [`\u{1F680} Framework: ${import_picocolors2.default.cyan(frameworkLabels[setupAnswers.framework])}`];
  if (setupAnswers.framework === "react" || setupAnswers.framework === "react-native" || setupAnswers.framework === "next") {
    summaryLines.push(`\u{1F3A8} Tailwind CSS: ${setupAnswers.useTailwind ? import_picocolors2.default.green("\u2713 Yes") : import_picocolors2.default.gray("\u2717 No")}`);
    summaryLines.push(`\u{1F4DA} Storybook: ${setupAnswers.useStorybook ? import_picocolors2.default.green("\u2713 Yes") : import_picocolors2.default.gray("\u2717 No")}`);
  }
  summaryLines.push(`\u{1F4C2} TypeScript Alias: ${setupAnswers.useTypescriptAlias ? import_picocolors2.default.green("\u2713 Yes") : import_picocolors2.default.gray("\u2717 No")}`);
  import_consola2.default.box({
    title: "\u{1F4CB} Configuration Summary",
    message: summaryLines.join("\n"),
    style: {
      borderColor: "blue",
      borderStyle: "rounded",
      padding: 1
    }
  });
  const packages = getPackagesToInstall(setupAnswers);
  import_consola2.default.info(`
\u{1F4E6} ${import_picocolors2.default.cyan(`${packages.length} packages`)} will be installed:
`);
  const gitHookPackages = ["husky", "lint-staged"];
  const commitlintPackages = ["@commitlint/cli", "@commitlint/config-conventional"];
  const eslintAndPrettierPackages = [
    "eslint",
    "prettier",
    "globals",
    "@eslint/js",
    "typescript-eslint",
    "eslint-config-prettier",
    "eslint-plugin-autofix",
    "eslint-plugin-import",
    "eslint-plugin-prefer-arrow-functions",
    "eslint-plugin-react",
    "eslint-plugin-react-hooks",
    "eslint-plugin-react-native",
    "@next/eslint-plugin-next",
    "eslint-plugin-tailwindcss",
    "prettier-plugin-tailwindcss",
    "eslint-plugin-storybook"
  ];
  const installedGitHookPackages = packages.filter((p) => gitHookPackages.includes(p));
  const installedCommitlint = packages.filter((p) => commitlintPackages.includes(p));
  const installedEslintPrettier = packages.filter((p) => eslintAndPrettierPackages.includes(p));
  if (installedGitHookPackages.length > 0) {
    import_consola2.default.info(import_picocolors2.default.bold("  \u{1FA9D} Git Hook Packages:"));
    installedGitHookPackages.forEach((pkg) => console.log(`     ${import_picocolors2.default.green("\u2022")} ${import_picocolors2.default.white(pkg)}`));
  }
  if (installedCommitlint.length > 0) {
    import_consola2.default.info(import_picocolors2.default.bold("\n  \u{1F4DD} Commitlint Packages:"));
    installedCommitlint.forEach((pkg) => console.log(`     ${import_picocolors2.default.yellow("\u2022")} ${import_picocolors2.default.white(pkg)}`));
  }
  if (installedEslintPrettier.length > 0) {
    import_consola2.default.info(import_picocolors2.default.bold("\n  \u{1F3A8} ESLint & Prettier Packages:"));
    installedEslintPrettier.forEach((pkg) => console.log(`     ${import_picocolors2.default.cyan("\u2022")} ${import_picocolors2.default.white(pkg)}`));
  }
  console.log("");
  const { confirmInstall } = await (0, import_prompts.default)({
    type: "confirm",
    name: "confirmInstall",
    message: "\u2728 Proceed with installation?",
    initial: true
  });
  if (!confirmInstall) {
    import_consola2.default.warn("Installation cancelled");
    return;
  }
  const pm = detectPackageManager();
  const pmIcons = {
    npm: "\u{1F4E6}",
    yarn: "\u{1F9F6}",
    pnpm: "\u{1F4CC}",
    bun: "\u{1F95F}"
  };
  import_consola2.default.start(`${pmIcons[pm] || "\u{1F4E6}"} Detected package manager: ${import_picocolors2.default.cyan(pm)}`);
  try {
    if (setupAnswers.useTypescriptAlias) {
      setupTypescriptAlias();
    }
    import_consola2.default.box({
      title: "\u2728 Setup Complete",
      message: [
        `${import_picocolors2.default.green("\u2713")} All development tools have been configured!`,
        "",
        `${import_picocolors2.default.cyan("What was set up:")}`,
        `  \u{1F4E6} ${import_picocolors2.default.bold("Packages")} - All dependencies installed`,
        `  \u{1F4DD} ${import_picocolors2.default.bold("Commitlint")} - Commit message validation`,
        `  \u{1F50D} ${import_picocolors2.default.bold("ESLint")} - Code linting`,
        `  \u{1F485} ${import_picocolors2.default.bold("Prettier")} - Code formatting`,
        `  \u{1FA9D} ${import_picocolors2.default.bold("Husky")} - Git hooks`,
        "",
        `${import_picocolors2.default.cyan("Next steps:")}`,
        `  ${import_picocolors2.default.gray("1.")} Start coding in your project`,
        `  ${import_picocolors2.default.gray("2.")} Commit with: ${import_picocolors2.default.green('git commit -m "feat: your feature"')}`,
        `  ${import_picocolors2.default.gray("3.")} Hooks will automatically validate and format your code`,
        "",
        `${import_picocolors2.default.gray("Example:")} ${import_picocolors2.default.green('git commit -m "feat: add awesome feature"')}`
      ].join("\n"),
      style: {
        borderColor: "green",
        borderStyle: "rounded",
        padding: 1
      }
    });
  } catch (error) {
    import_consola2.default.error("Failed to install packages");
    if (error instanceof Error) {
      import_consola2.default.error(error.message);
    }
    process.exit(1);
  }
};

// src/scripts/cli.ts
import_commander.program.name("node-devtools").description("Interactive devtools (commitlint, eslint, prettier,...) setup for Node.js projects").version("1.2.0");
import_commander.program.description("Interactive setup - Install and configure all dev dependencies").action(async () => {
  await runInteractiveSetup();
});
import_commander.program.parse();
