// @jjuidev/node-devtools/dist/cjs/index.cjs
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var index_exports = {};
__export(index_exports, {
  commitlintConfigRecommend: () => commitlintConfigRecommend,
  defineCommitlintConfig: () => defineCommitlintConfig,
  defineEslintConfig: () => defineEslintConfig,
  eslintConfigNode: () => eslintConfigNode,
  pluginNext: () => pluginNext,
  pluginReact: () => pluginReact,
  pluginStorybook: () => pluginStorybook
});
module.exports = __toCommonJS(index_exports);

// src/packages/commitlint/index.ts
var createCommitlintConfigRecommend = (commitlintEmoji) => {
  return {
    // https://commitlint.js.org/#/reference-configuration
    extends: ["@commitlint/config-conventional"],
    rules: {
      "type-enum": [2, "always", commitlintEmoji.map((item) => item.type)],
      "type-case": [2, "always", "lower-case"],
      "type-empty": [2, "never"],
      "subject-case": [2, "never"],
      "subject-empty": [2, "never"],
      "subject-min-length": [2, "always", 10],
      "subject-max-length": [2, "always", 120]
    },
    // https://github.com/folke/devmoji
    types: commitlintEmoji.map((item) => item.type),
    devmoji: commitlintEmoji.map(({ code, emoji, description }) => ({
      code,
      emoji,
      description
    }))
  };
};
var COMMITLINT_EMOJI_RECOMMEND = [
  {
    type: "init",
    code: ":init:",
    emoji: ":tada:",
    description: "Initialize project"
  },
  {
    type: "feat",
    code: ":feat:",
    emoji: ":sparkles:",
    description: "Add new feature"
  },
  {
    type: "fix",
    code: ":fix:",
    emoji: ":bug:",
    description: "Fix a bug"
  },
  {
    type: "chore",
    code: ":chore:",
    emoji: ":wrench:",
    description: "Minor tasks or maintenance"
  },
  {
    type: "docs",
    code: ":docs:",
    emoji: ":memo:",
    description: "Update documentation"
  },
  {
    type: "style",
    code: ":style:",
    emoji: ":lipstick:",
    description: "Improve UI or code style"
  },
  {
    type: "improve",
    code: ":improve:",
    emoji: ":rocket:",
    description: "Improve code quality or performance or readability or maintainability or etc"
  },
  {
    type: "refactor",
    code: ":refactor:",
    emoji: ":recycle:",
    description: "Refactor code without changing logic"
  },
  {
    type: "perf",
    code: ":perf:",
    emoji: ":zap:",
    description: "Enhance performance"
  },
  {
    type: "test",
    code: ":test:",
    emoji: ":white_check_mark:",
    description: "Add or update tests"
  },
  {
    type: "build",
    code: ":build:",
    emoji: ":building_construction:",
    description: "Changes related to the build system"
  },
  {
    type: "ci",
    code: ":ci:",
    emoji: ":repeat:",
    description: "Configure CI/CD"
  },
  {
    type: "revert",
    code: ":revert:",
    emoji: ":rewind:",
    description: "Revert a previous commit"
  },
  {
    type: "merge",
    code: ":merge:",
    emoji: ":twisted_rightwards_arrows:",
    description: "Merge branches"
  },
  {
    type: "wip",
    code: ":wip:",
    emoji: ":construction:",
    description: "Work in progress"
  },
  {
    type: "release",
    code: ":release:",
    emoji: ":rocket:",
    description: "Release a new version"
  },
  {
    type: "upgrade",
    code: ":upgrade:",
    emoji: ":arrow_up:",
    description: "Upgrade dependencies or software"
  },
  {
    type: "downgrade",
    code: ":downgrade:",
    emoji: ":arrow_down:",
    description: "Downgrade dependencies or software"
  },
  {
    type: "bump",
    code: ":bump:",
    emoji: ":package:",
    description: "Bump package version"
  },
  {
    type: "security",
    code: ":security:",
    emoji: ":lock:",
    description: "Improve security"
  },
  {
    type: "hotfix",
    code: ":hotfix:",
    emoji: ":fire:",
    description: "Urgent bug fix"
  },
  {
    type: "maintainer",
    code: ":maintainer:",
    emoji: ":crown:",
    description: "Maintainer commit and excellent handle for system"
  }
];
var commitlintConfigRecommend = createCommitlintConfigRecommend(COMMITLINT_EMOJI_RECOMMEND);
var defineCommitlintConfig = (configFn) => {
  if (!configFn) {
    return commitlintConfigRecommend;
  }
  const emojiList = configFn(COMMITLINT_EMOJI_RECOMMEND);
  return createCommitlintConfigRecommend(emojiList);
};

// src/packages/eslint/index.ts
var import_js = __toESM(require("@eslint/js"), 1);
var import_eslint_plugin_autofix = __toESM(require("eslint-plugin-autofix"), 1);
var import_eslint_plugin_import = __toESM(require("eslint-plugin-import"), 1);
var import_eslint_plugin_prefer_arrow_functions = __toESM(require("eslint-plugin-prefer-arrow-functions"), 1);
var import_config = require("eslint/config");
var import_globals = __toESM(require("globals"), 1);
var import_typescript_eslint = __toESM(require("typescript-eslint"), 1);
var loadModule = async (name) => {
  try {
    const module2 = await import(name);
    return module2.default || module2;
  } catch (err) {
    console.error(`\u274C Failed to load module: ${name}`);
    console.error(`   Please install it: npm install --save-dev ${name}`);
    throw err;
  }
};
var defineEslintConfig = async (...configs) => {
  const resolvedConfigs = await Promise.all(
    configs.map(async (config) => {
      if (config instanceof Promise) {
        return await config;
      }
      if (Array.isArray(config)) {
        return config;
      }
      return config;
    })
  );
  return (0, import_config.defineConfig)(...resolvedConfigs.flat());
};
var eslintConfigNode = defineEslintConfig(
  {
    files: ["**/*.{js,ts,jsx,tsx}"],
    languageOptions: {
      globals: {
        ...import_globals.default.node,
        ...import_globals.default.browser
      }
    }
  },
  import_js.default.configs.recommended,
  import_typescript_eslint.default.configs.recommended,
  {
    plugins: {
      import: import_eslint_plugin_import.default,
      "prefer-arrow-functions": import_eslint_plugin_prefer_arrow_functions.default,
      autofix: import_eslint_plugin_autofix.default
    },
    rules: {
      // autofix
      "autofix/eol-last": "error",
      "autofix/curly": "error",
      "autofix/no-lonely-if": "error",
      "autofix/no-else-return": "error",
      "autofix/object-curly-newline": [
        "error",
        {
          ObjectExpression: {
            multiline: true,
            minProperties: 2,
            consistent: true
          }
        }
      ],
      "object-shorthand": "warn",
      // @typescript-eslint
      "@typescript-eslint/no-unused-vars": "warn",
      "@typescript-eslint/no-explicit-any": "off",
      "@typescript-eslint/no-empty-object-type": "off",
      // eslint-plugin-prefer-arrow-functions
      "prefer-arrow-functions/prefer-arrow-functions": "error",
      // eslint-plugin-import
      "import/first": "error",
      "import/newline-after-import": "error",
      "import/no-duplicates": "error",
      "import/no-anonymous-default-export": "error",
      "import/order": [
        "error",
        {
          "newlines-between": "always",
          alphabetize: {
            order: "asc",
            caseInsensitive: true
          },
          pathGroups: [
            {
              pattern: "@/**",
              group: "internal"
            }
          ],
          groups: ["builtin", "external", ["internal", "parent", "sibling", "index"], ["object", "unknown", "type"]]
        }
      ],
      // Others rules
      "padding-line-between-statements": [
        "error",
        {
          blankLine: "any",
          prev: "export",
          next: "export"
        },
        {
          blankLine: "always",
          prev: ["const", "let", "var"],
          next: "*"
        },
        {
          blankLine: "any",
          prev: ["const", "let", "var"],
          next: ["const", "let", "var"]
        },
        {
          blankLine: "always",
          prev: "*",
          next: ["function", "multiline-const", "multiline-block-like"]
        },
        {
          blankLine: "always",
          prev: ["function", "multiline-const", "multiline-block-like"],
          next: "*"
        }
      ]
    }
  }
);
var pluginReact = () => {
  return (async () => {
    const [pluginReact2, pluginReactHooks, pluginReactNative] = await Promise.all([
      loadModule("eslint-plugin-react"),
      loadModule("eslint-plugin-react-hooks"),
      loadModule("eslint-plugin-react-native")
    ]);
    return defineEslintConfig({
      settings: {
        react: {
          version: "detect"
        }
      },
      plugins: {
        react: pluginReact2,
        "react-native": pluginReactNative,
        "react-hooks": pluginReactHooks
      },
      rules: {
        // eslint-plugin-react
        "react/jsx-uses-react": "off",
        "react/react-in-jsx-scope": "off",
        "react/display-name": "off",
        "react/jsx-boolean-value": "error",
        "react/jsx-curly-brace-presence": ["error", "never"],
        "react/self-closing-comp": "error",
        // eslint-plugin-react-native
        "react-native/no-unused-styles": "warn",
        "react-native/no-inline-styles": "warn",
        "react-native/no-color-literals": "warn",
        "react-native/no-raw-text": "off",
        // eslint-plugin-react-hooks
        "react-hooks/rules-of-hooks": "error",
        "react-hooks/exhaustive-deps": "warn"
      }
    });
  })();
};
var pluginNext = () => {
  return (async () => {
    const pluginNext2 = await loadModule("@next/eslint-plugin-next");
    return defineEslintConfig(pluginReact(), {
      plugins: {
        "@next/next": pluginNext2
      },
      rules: {
        // eslint-plugin-next
        ...pluginNext2.configs.recommended.rules,
        ...pluginNext2.configs["core-web-vitals"].rules
      }
    });
  })();
};
var pluginStorybook = () => {
  return (async () => {
    const pluginStorybook2 = await loadModule("eslint-plugin-storybook");
    return [
      {
        plugins: {
          storybook: pluginStorybook2
        },
        rules: {
          // eslint-plugin-storybook
          ...pluginStorybook2.configs.recommended.rules
        }
      }
    ];
  })();
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  commitlintConfigRecommend,
  defineCommitlintConfig,
  defineEslintConfig,
  eslintConfigNode,
  pluginNext,
  pluginReact,
  pluginStorybook
});
